processor/src/main/java/org/mapstruct/ap/internal/util/Message.java

```diff
@@ -179,7 +179,11 @@
     VALUEMAPPING_ANY_REMAINING_OR_UNMAPPED_MISSING( "Source = \"<ANY_REMAINING>\" or \"<ANY_UNMAPPED>\" is advisable for mapping of type String to an enum type.", Diagnostic.Kind.WARNING  ),
     VALUEMAPPING_NON_EXISTING_CONSTANT_FROM_SPI( "Constant %s doesn't exist in enum type %s. Constant was returned from EnumMappingStrategy: %s"),
     VALUEMAPPING_NON_EXISTING_CONSTANT( "Constant %s doesn't exist in enum type %s." ),
-    VALUEMAPPING_THROW_EXCEPTION_SOURCE( "Source = \"<THROW_EXCEPTION>\" is not allowed. Target = \"<THROW_EXCEPTION>\" can only be used." );
+    VALUEMAPPING_THROW_EXCEPTION_SOURCE( "Source = \"<THROW_EXCEPTION>\" is not allowed. Target = \"<THROW_EXCEPTION>\" can only be used." ),
+
+    MAPTOBEANMAPPING_WRONG_GENERIC_TYPES( "The Map parameter \"%s\" cannot be used for property mapping. It must be typed with Map<String, ???> but it was typed with %s.", Diagnostic.Kind.WARNING ),
+    MAPTOBEANMAPPING_UNTYPED( "The Map parameter \"%s\" cannot be used for property mapping. It must be typed with Map<String, ???> but it was untyped.", Diagnostic.Kind.WARNING ),
```

Comment:
I guess this one is used only when you try to do a raw map. Shouldn't the message reflect that?. So something along the lines `Raw Map is not allowed`

Reply:
Yes this is for raw maps. I'll adapt the warning message.

Reply:
Just saw that that message ends with "but it was untyped". I think that it is OK to do it like that. I'll change it tp "but it was raw"